build/yamls/flow-visibility/base/clickhouse.yml

```diff
@@ -26,7 +26,24 @@ spec:
   defaults:
     templates:
       podTemplate: pod-template
+      hostTemplate: host-template
```

Comment:
curious why do we need this now? previously the operator can create the related Service without these templates. I'm assuming there's some merge issues on operator side to expose the ports on the correct container: https://github.com/Altinity/clickhouse-operator/blob/cecb54ad1ade33a9d748a0108e765abad9b2412e/deploy/operator/clickhouse-operator-install-bundle.yaml#L3510 
maybe we can try manually add these and see if it works?

Reply:
This is added for allowing customized port as mentioned in the discussion above: https://github.com/antrea-io/antrea/pull/3498#discussion_r833857276

I'm not quite understand the merge issues you mention. But I've tried to change the port number in the hostTemplate, and it can correctly change the corresponding port exposed by ClickHouse server container in the pod `chi-clickhouse-clickhouse-0-0`, which is used by `clickhouse client` commands inside the container. 

I also have a quick try on adding ports in the ClickHouse container only. It seems that change the port number does not actually change the port exposed by the pod. With some investigation I think this might be just informative as stated [here](https://stackoverflow.com/questions/57197095/why-do-we-need-a-port-containerport-in-a-kuberntes-deployment-container-definiti).

Reply:
ok understood. and right, to change server ports hostTemplate is the source of truth.